assembly-plugin-boilerplate/README.md

```diff
@@ -0,0 +1,10 @@
+# Assembly Plugin Docker Boilerplate
+
+This project uses the assembly plugin rules from resource.xml to package
+an artifact containing the raw contents of package.xml. See src/assembly,
```

Comment:
historically we have `package.xml`, `development.xml`, and `standalone.xml`. the standalone one was mainly for testing convenience as it introduces redistribution issues. development includes enough to test in a dev environment.

i mention this because the merging of docker stuff makes it unclear how this is intended to interact with traditional packaging and the dev cycle. how would this affect those other targets for downstream? is packaging going to be affected in anyway by including this in common?

Reply:
@aayars @ewencp - We believe it wont affect, however this needs to be validated once the merge goes in. I spoke to Alex about this, in case we run into issues, we intend to revert it back.

Reply:
This is a good point, and probably worth nailing down as part of this PR. `rest-utils` brings in the assembly configs mentioned by @ewencp, and there is a fair amount of overlap between the "docker" assemblies and the "packaging" assemblies. There's also overlap with the license generation stuff between `common-docker` and `rest-utils`.

For projects which inherit from `rest-utils` I am using rest-utils' assemblies (used by packaging) instead of common-docker's... just because they were already there, and happen to be perfectly suitable for copying into an image.

I propose consolidating the assembly configs from `rest-utils` into `common`. We can update the Dockerfile in downstream projects to use that style of assembly (including license file generation stuff), instead of the `common-docker` style of assembly. It's possible that there's no change at all required for some projects.